File: Q2 sort the colors.py
class Solution(object):
    def sortColors(self, nums):
      n=len(nums) 
      right=n-1
      index,left=0,0
      while(index<=right):
        if(nums[index]==0):
          temp=nums[index]
          nums[index]=nums[left]
          nums[left]=temp
          left=left+1
          index=index+1
        elif nums[index]==2:
            temp=nums[index]
            nums[index]=nums[right]
            nums[right]=temp
            right=right-1
        else:
         index=index+1
        

File: Q2 sort the colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var sortColors = function(nums) {
    var n = nums.length;
    var index = 0;
    var left = 0;
    var right = n - 1; 

    while (index <= right) {
        if (nums[index] === 0) {
            var temp = nums[index];
            nums[index] = nums[left];
            nums[left] = temp;
            left++;
            index++;
        } else if (nums[index] === 2) {
            var temp = nums[index];
            nums[index] = nums[right];
            nums[right] = temp;
            right--;
        } else {
            index++;
        }
    }
};
